cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -253,6 +324,34 @@ public void error(Throwable cause) {
     }
   }
 
+  @Nullable
+  private ApplicationSpecification regenerateAppSpec(
+    Map<String, String> systemArguments, Map<String, String> userArguments, ProgramId programId,
+    ApplicationSpecification existingAppSpec, ProgramDescriptor programDescriptor,
+    ConfiguratorFactory configuratorFactory) throws InterruptedException, ExecutionException, TimeoutException {
+
+    String appClassName = systemArguments.get(ProgramOptionConstants.APPLICATION_CLASS);
+    Location programJarLocation = Locations.toLocation(
+      new File(systemArguments.get(ProgramOptionConstants.PROGRAM_JAR)));
+
+
+    AppDeploymentInfo deploymentInfo = new AppDeploymentInfo(
+      programDescriptor.getArtifactId(), programJarLocation, programId.getNamespaceId(), appClassName,
+      programId.getApplication(), programId.getVersion(), existingAppSpec.getConfiguration(), null, false,
+      new AppDeploymentRuntimeInfo(existingAppSpec, userArguments, systemArguments));
+    Configurator configurator = configuratorFactory.create(deploymentInfo);
+    ListenableFuture<ConfigResponse> future = configurator.config();
+    ConfigResponse response = future.get(120, TimeUnit.SECONDS);
```

Comment:
Should this value be configurable via the cdap-site.xml?

Reply:
This exists long time for our app deploy, created https://cdap.atlassian.net/browse/CDAP-18777 for an improvement